go/common/crypto/signature/signers/ledger/ledger_signer.go

```diff
@@ -0,0 +1,85 @@
+package ledger
+
+import (
+	"encoding/binary"
+
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/oasislabs/ledger-go"
+	"github.com/oasislabs/oasis-core/go/common/crypto/signature"
+)
+
+const (
+	ETH_CLA        byte = 0xE0
+	ETH_INS_CONFIG byte = 0x06
+	ETH_INS_DERIVE byte = 0x02
+	ETH_INS_SIGN   byte = 0x04
+)
+
+type Signer struct {
+	device *ledger_go.Ledger
+}
+
+// NewSigner will attempt to initialize a Ledger Signer by running
+// FindLedger. This will connect to the first Ledger device found.
+func NewSigner() (*Signer, error) {
+	device, err := ledger_go.FindLedger()
+	if err != nil {
+		return nil, err
+	}
+	return &Signer{device}, nil
+}
+
+func (s *Signer) Public() signature.PublicKey {
+	derivationPath := accounts.DefaultRootDerivationPath
+	path := make([]byte, 1+4*len(derivationPath))
+	path[0] = byte(len(derivationPath))
+	for i, component := range derivationPath {
+		binary.BigEndian.PutUint32(path[1+4*i:], component)
+	}
+
+	header := []byte{ETH_CLA, ETH_INS_DERIVE, 0, 0, byte(len(path))}
+	message := append(header, path...)
+
+	response, err := s.device.Exchange(message)
```

Comment:
Is the public key expected to change between invocations? Otherwise the signer should cache this instead of talking to the device each time a public key is requested.